exercises/practice/palindrome-products/tests/palindrome-products.rs

```diff
@@ -3,79 +3,93 @@
 
 use palindrome_products::{palindrome_products, Palindrome};
 
+
 /// Process a single test case for the property `smallest`
 ///
 /// All cases for the `smallest` property are implemented
 /// in terms of this function.
-fn process_smallest_case(input: (u64, u64), expected: Option<Palindrome>) {
-    let min = palindrome_products(input.0, input.1).map(|(min, _)| min);
-    assert_eq!(min, expected);
+fn process_smallest_case((from, to): (u64, u64), expected: Option<u64>) {
+    let min = palindrome_products(from, to).map(|(min, _)| min);
+    assert_eq!(min.map(|newtype| newtype.into_inner()), expected);
 }
 
 /// Process a single test case for the property `largest`
 ///
 /// All cases for the `largest` property are implemented
 /// in terms of this function.
-fn process_largest_case(input: (u64, u64), expected: Option<Palindrome>) {
-    let max = palindrome_products(input.0, input.1).map(|(_, max)| max);
-    assert_eq!(max, expected);
+fn process_largest_case((from, to): (u64, u64), expected: Option<u64>) {
+    let max = palindrome_products(from, to).map(|(_ , max)| max);
+    assert_eq!(max.map(|newtype| newtype.into_inner()), expected);
 }
 
 #[test]
+/// test the Palindrome's new method, a valid input
```

Comment:
```suggestion
/// test `Palindrome::new` with valid input
```

Just using more conventional notation to unambiguously identify the method.